src/main/java/domain/LadderGame.java

```diff
@@ -20,4 +26,26 @@ public Names getNames() {
     public Ladder getLadder() {
         return ladder;
     }
+
+    public LadderResults getLadderResults() {
+        return ladderResults;
+    }
+
+    public LadderGameResult calculateLadderGameResult() {
+        ladder.getLadder().stream()
+                .map(Bridges::getBridges)
+                .forEach(this::swapNames);
+        return new LadderGameResult(names, ladderResults);
+    }
+
+    private void swapNames(List<Boolean> bridges) {
+        IntStream.range(0, bridges.size())
+                .forEach(index -> swapIfBridgeExist(bridges.get(index), index));
+    }
+
+    private void swapIfBridgeExist(Boolean bridge, int index) {
```

Comment:
```suggestion
    private void swapIfBridgeExist(boolean bridge, int index) {
```


흥미로운 구현이군요. 잘 구현하신 것 같은데, names의 상태를 바꾸기보다 새로운 names를 만들게 하면 어떤가요?

```java
// swapNames
forEach > map

private Names swapIfBridgeExist(boolean bridge, int index) {
    ...
    return names.swapNamePosition(index + 1);
}

```

Reply:
다시보니 진짜 swap이 아니라 임시 저장소에 스냅샷을 넣는거군요.

그럼 Names의 역할과는 거리가 있어 보이는데, 따로 객체로 분리해보면 어떤가요?

Reply:
처음 구현할 땐, `Names`가 참여자 이름들을 저장하고 있어서 위치를 변경하는 데에는 문제가 없어 보였어요.

그러나, 참여자 이름 위치를 변경하는 것은 사다리 게임에서 필요한 것이지, 참여자 이름들 자체에선 필요없는 기능이었고, `Names`가 사다리 게임에 의존하는 것처럼 보이게 되었네요.

이름 위치를 변경하는 역할은 LadderGameResult가 가져가는 편이 좋다고 생각합니다!

Reply:
> 이름 위치를 변경하는 역할은

변경하는 역할과 변경해야 할 내용을 만드는 역할은 다를수도 있을것 같아요. 또는 **변경이 아니라 새로 생성으로** 구현한다면, `변경` 이라는 개념이 필요하지 않을지도 모르고요.

Reply:
오래 고민한 결과, 이부분은 새로운 도메인이 추가 될 것 같네요.
가령, `참여자 개수`를 `사다리`에 전달해서 변경 할 사다리 위치를 생성 할 수 도 있겠어요!